GVFS/GVFS.Platform.Mac/ProcessLauncher.cs

```diff
@@ -89,6 +94,15 @@ private static string[] GenerateArgv(string fileName, string[] args)
             return argvList.ToArray();
         }
 
+        private static void LogErrorAndExit(ITracer tracer, string message)
+        {
+            int lastWin32Error = Marshal.GetLastWin32Error();

```

Comment:
`Marshal.GetLastWin32Error()` is now quite a bit farther away from where the error is generated. We have to be careful to not run any action that might clear the last error before running this method. I think this would be harder to inadvertently break if the GetLastWin32Error() was closer to the native call... Maybe at least add a comment to call out this requirement in code to remind future people to keep this mind? (assuming my understanding of GetLastWin32Error is correct)

Reply:
I've got the same concern, what if we made the native error code a parameter to `LogErrorAndExit` and continued to call `GetLastWin32Error` before entering this method?

Reply:
Will do.